TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -99,38 +113,59 @@ public LaneConnectorTool(TrafficManagerTool mainTool)
             Vector3 camPos = Singleton<SimulationManager>.instance.m_simulationView.m_position;
 
             // Bounds bounds = new Bounds(Vector3.zero, Vector3.one);
-            Ray mouseRay = Camera.main.ScreenPointToRay(Input.mousePosition);
+            Camera currentCamera = Camera.main;
+            Ray mouseRay = currentCamera.ScreenPointToRay(Input.mousePosition);
+
+            // Check if camera pos/angle has changed then re-filter the visible nodes
+            // Assumption: The states checked in this loop don't change while the tool is active
```

Comment:
What about overlay? Tool doesn't have to be enabled to render its overlay (`viewOnly`).
I agree that for now this kind of optimization is OK, but we need something better in the future.

Reply:
This might potentially cache too much, if the user has overlay on and is editing roads. But moving the camera will have it fixed right away. Also isn't this always view overlay for debug only?

Reply:
You can't have overlay on and edit/build road ;)

Btw I've noticed that overlay is not rendered when you deselect and select tool without moving camera afterwards

Reply:
> Btw I've noticed that overlay is not rendered when you deselect and select tool without moving camera afterwards

I was unable to reproduce this but will do some more testing.

Reply:
Speed limits tool only

Reply:
Added cache reset when the tool is activated, in all 3 affected tools. Testing...

Reply:
The overlays should be rendered by other code than the tool itself. I.e. having the tool code called when the tool is not active is terrible mistake and the render overlay and caching code should be moved outside. Now I also observe that when Overlay is on and Speed limits tool is on and off, the speed limits disappear until the camera is moved.

Reply:
I've now been able to replicate the issue with speed limits.

Note to self; Camera moves when it's over water. lol